src/nextjournal/clerk/parser.cljc

```diff
@@ -164,17 +170,20 @@
                                  :nextjournal.clerk/visibility {:code :fold}}(inc 1))
 
 (defn ->doc-settings [first-form]
-  {:ns? (ns? first-form)
-   :error-on-missing-vars (parse-error-on-missing-vars first-form)
-   :toc-visibility (or (#{true :collapsed} (:nextjournal.clerk/toc
-                                            (merge (-> first-form meta) ;; TODO: deprecate
-                                                   (when (ns? first-form)
-                                                     (merge (-> first-form second meta)
-                                                            (first (filter map? first-form)))))))
-                       false)
-   :block-settings (merge-with merge
-                               {:nextjournal.clerk/visibility {:code :show :result :show}}
-                               (parse-global-block-settings first-form))})
+  (merge
+   (when (ns? first-form) (select-keys (first (filter map? first-form)) doc-settings))
+   {:ns? (ns? first-form)
+    :error-on-missing-vars (parse-error-on-missing-vars first-form)
+    ;; TODO: unify with doc-settings, expose :n.c/toc-visibility user-facing
+    :toc-visibility (or (#{true :collapsed} (:nextjournal.clerk/toc
```

Comment:
We should either leave all doc keys namespaced or none so `:doc-css-class` is treated the same as `:toc-visibility`.

Reply:
I'd be for either adding `:nextjournal.clerk/toc-visibility` to the list and exposing it user facing instead of `:n.c./toc` or  use `:n.c/toc` throughout the app and have it land to the render function as is and interpret it as _toc visibility_. Won't conflict with the unqualified `:toc`.

I'm also assusming we don't need the key-rename to `:nextjournal/X` for these keys.